scripts/core/editor3/actions/spellchecker.jsx

```diff
@@ -21,14 +16,61 @@ export function replaceWord(word) {
 
 /**
  * @ngdoc method
- * @name Spellchecker Actions#showContextMenu
- * @param {Object} contextMenuData
+ * @name replaceWord
+ * @param {String} word
+ * @return {String} action
+ * @description Creates the add word action
+ */
+export function addWord(word) {
+    const spellcheck = ng.get('spellcheck');
+
+    spellcheck.addWord(word, false);
+
+    return replaceWord(word);
+}
+
+/**
+ * @ngdoc method
+ * @name replaceWord
+ * @param {String} word
+ * @return {String} action
+ * @description Creates the ignore word action
+ */
+export function ignoreWord(word) {
+    const spellcheck = ng.get('spellcheck');
+
+    spellcheck.addWord(word, true);
+
+    return replaceWord(word);
```

Comment:
Here also I don't know how to implement an alternative solution. It would have to trigger the editor's change event for it to re-run the decorators. We could look at it as "_replace the word with the new word that is now in the dictionary_" :-)

Also open to suggestions here...

I have a similar issue somewhere else and if I figure out how to do this I will change this too.